packages/core/src/browser/theming.ts

```diff
@@ -0,0 +1,156 @@
+
+    private themes: { [id: string]: Theme } = {};
+    private activeTheme: Theme | undefined;
+    themeChange = new Emitter<ThemeChangeEvent>();
+
+    protected constructor(private defaultTheme: string) { }
+
+    register(theme: Theme) {
+        this.themes[theme.id] = theme;
+    }
+
+    getThemes(): Theme[] {
+        const result = [];
+        for (const o in this.themes) {
+            if (this.themes.hasOwnProperty(o)) {
+                result.push(this.themes[o]);
+            }
+        }
+        return result;
+    }
+
+    setCurrentTheme(themeId: string) {
+        const newTheme = this.themes[themeId] || this.themes[this.defaultTheme];
+        const oldTheme = this.activeTheme;
+        if (oldTheme) {
+            oldTheme.deactivate();
+        }
+        newTheme.activate();
+        this.activeTheme = newTheme;
+        this.themeChange.fire({
+            newTheme, oldTheme
+        });
+        window.localStorage.setItem('theme', themeId);
+    }
+
+    getCurrentTheme(): Theme {
+        const themeId = window.localStorage.getItem('theme') || this.defaultTheme;
+        return this.themes[themeId] || this.themes[this.defaultTheme];
+    }
+
+    static get() {
+        // tslint:disable-next-line:no-any
+        const wnd = window as any;
+        if (!wnd.__themeService) {
+            const themeService = new ThemeService('dark');
+            wnd.__themeService = themeService;
+        }
+        return wnd.__themeService as ThemeService;
+    }
+}
+ThemeService.get().register(darkTheme);
+ThemeService.get().register(lightTheme);
+
+@injectable()
+export class ThemingCommandContribution implements CommandContribution, CommandHandler, Command, QuickOpenModel {
+
+    id = 'change_theme';
+    label = 'Change Color Theme';
+
+    constructor( @inject(QuickOpenService) protected openService: QuickOpenService) { }
+
+    registerCommands(commands: CommandRegistry): void {
+        commands.registerCommand(this, this);
+    }
+
+    execute() {
+        this.openService.open(this, {
+            placeholder: 'Select Color Theme (Up/Down Keys to Preview)',
+            fuzzyMatchLabel: true,
+            selectIndex: () => this.activeIndex()
+        });
+    }
+
+    private activeIndex() {
+        const themeService = ThemeService.get();
+        const current = themeService.getCurrentTheme().id;
+        const themes = themeService.getThemes();
+        for (let i = 0; i < themes.length; i++) {
+            if (themes[i].id === current) {
+                return i;
+            }
+        }
+        return -1;
+    }
+
+    onType(lookFor: string, acceptor: (items: QuickOpenItem[]) => void): void {
+        const themeService = ThemeService.get();
+        const items = themeService.getThemes().map(t =>
+            new QuickOpenItem({
+                label: t.label,
+                description: t.description,
+                run(mode: QuickOpenMode) {
+                    themeService.setCurrentTheme(t.id);
```

Comment:
I wonder if I preview one theme and then close the quick panel will it rollback the current theme?

Reply:
good catch. fixed it